src/Codeception/Util/ReflectionPropertyAccessor.php

```diff
@@ -47,29 +47,23 @@ private function setPropertiesForClass($obj, $class, array $data)
     {
         $reflectedEntity = new ReflectionClass($class);
 
-        $constructorParameters = [];
-        $constructor = $reflectedEntity->getConstructor();
-        if (null !== $constructor) {
-            foreach ($constructor->getParameters() as $parameter) {
-                if ($parameter->isOptional()) {
-                    $constructorParameters[] = $parameter->getDefaultValue();
-                } elseif (array_key_exists($parameter->getName(), $data)) {
-                    $constructorParameters[] = $data[$parameter->getName()];
-                } else {
-                    throw new InvalidArgumentException(
-                        'Constructor parameter "'.$parameter->getName().'" missing'
-                    );
-                }
-            }
+        /*
+         * In case we need to create the entity via reflection, we need to have all constructor parameters
+         * in the $data array, in case we only want to enhance an existing instance, the given $obj has
+         * already been constructed with all of its parameters outside of this function,
+         * so we can skip creating it via reflection.
+         */
+        if ($this->isNoObjectOf($obj, $class)) {
```

Comment:
I'm not exactly sure what is going on here, as `setPropertiesForClass` is a private method and is called with either `$obj = null`, or `$class = get_class($obj)`. So, `$obj` is always a `null` or instance of `$class`, guaranteed. Why do we use `is_a` here, and why `is_a` is wrapped into separate method? Not saying this is wrong, just can't see the reasoning behind this.

Reply:
For me its the responsibility of `setPropertiesForClass` method itself to check whether we deal with an instantiated object of exactly this class and not the responsibility of the caller. So whenever anyone (in the future?) will call `setPropertiesForClass` with an object not instance of `$class`, it will create an instance of the given class instead.

The wrapper is just my understanding of clean code, so it reads better than any if clause with a `false` in it. (less cognitive load).

Reply:
As the one who wrote the method (and whole class), I assure you that this private method expects either `null` or object instance of specified class, by design -- this is a helper method, nothing more. Feel free to change name to a clearer one, or add a comment, but please do not change semantics without any reason and any comment.

Creating instance of given class instead (and discarding passed instance) is something I'd call "huge surprise". Throwing exception instead is much more clear behavior. Or, even better, leave original design alone.

Reply:
Re wrapper, you're wrapping function which checks if object is an instance of given class with a function which... does literally the same, without adding anything new. This *is* more cognitive load, because future reader will look into that function to see basically nothing.